client.go

```diff
@@ -59,8 +59,14 @@ func run() {
 		app.ui.echoerrf("reading history file: %s", err)
 	}
 
+	// enable focus reporting
+	fmt.Print("\x1b[?1004h")
+
 	app.loop()
 
+	// disable focus reporting
+	fmt.Print("\x1b[?1004l")
```

Comment:
This is actually a little buggy. Let's say you run `lf` inside another `lf`. After you exit the inner `lf`, the outer `lf` no longer receives focus events.

Also, if you use `lf` to launch a program that gets confused by focus events, that program will get focus events and be confused.

To solve this, every time `lf` launches a program that might have access to the terminal (so, `$` and `!` but not `%`), it should first disable focus reporting, and then enable focus reporting again after that program finishes.

Reply:
Thanks, that's a good input. focus events should be disabled in that case. I'll take a look when I find the time.

Reply:
You should also consider using Go's `defer` feature to attempt to disable focus events in cases when `lf` is interrupted or something panics. (I'm 80% sure it's the right thing to do, would need to do a bit more research to be sure).